### PR TITLE
Fixes 7478: splitLines keepEOL option

### PR DESCRIPTION
**Pull request by drslump**
Fixes https://github.com/nim-lang/Nim/issues/7478

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -649,22 +650,29 @@ iterator splitLines*(s: string): string =
...
   var last = 0
+  var eolpos = 0
   while true:
     while last < s.len and s[last] notin {'\c', '\l'}: inc(last)
-    yield substr(s, first, last-1)
-    # skip newlines:
-    if last >= s.len: break
+
+    if last >= s.len:
+      yield substr(s, first, last-1)
```
Comment (Araq) [inline lib/pure/strutils.nim]: These additional ``yield``'s can cause serious code bloat.

Comment (drslump) [reply lib/pure/strutils.nim]: I see, I'm not familiar with how Nim's generates the state machine to handle the generator state, in other languages I've worked with once you have a generator additional yields are not much of an issue, `except` blocks is usually what can significally increase the bloat.

I'll rework the logic to keep a single `yield` in the function.

**Comment by drslump** [comment]
Changes done so the generator keeps having a single `yield` expression (thanks @Araq). Also changed `keepEOL` to `keepEol`, I think I've read somewhere in the compiler docs that the later form is preferred (although can't find where exactly right now).

**Comment by Araq** [comment]
Mergeable once the CIs are green.

**Pull request merged**